vite-project/src/pages/WebSocketChat.tsx

```diff
@@ -0,0 +1,190 @@
+import { User, useAuth0 } from "@auth0/auth0-react";
+import React, { useState } from "react";
+import { Message, Users, useWebSocketChat } from "../hooks/useWebSocketChat";
+import Button from "../components/Button";
+import { Loader } from "../components";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { createPost } from "../lib/api";
+import { SinglePost, postData } from "../lib/types";
+import MuiLoader from "../components/MuiLoader";
+import Input from "../components/Input";
+
+const sortUsers = (userList: Users, chatUser: string): User[] => {
+  const result = Object.keys(userList)
+    .sort((a, b) => {
+      if (userList[a].user === chatUser) return -1;
+      if (userList[b].user === chatUser) return 1;
+      return 0;
+    })
+    .map((key) => userList[key]);
+  return result;
+};
+
+//TODO add functionality to chat with gpt in the chat window
+
+export default function WebSocketChat() {
+  const { user } = useAuth0();
+  const queryClient = useQueryClient();
+  const [sharedImagesArr, setSharedImagesArr] = useState<number[]>([]);
+
+  const {
+    imageLoading,
+    handleJoinChat,
+    handleSendMessage,
+    messageLog,
+    userList,
+    newMessage,
+    command,
+    setCommand,
+    additionalText,
+    chatUser,
+    setChatUser,
+    handleInputChange,
+    handleKeyDown,
+    isUserJoined,
+  } = useWebSocketChat(user);
+
+  const { mutate, isLoading } = useMutation<SinglePost, unknown, postData>(
+    (newData: postData) => createPost(newData),
+    {
+      onSuccess: () => {
+        //@ts-expect-error passing in an array does not work
+        queryClient.invalidateQueries("posts");
+      },
+      onError(error) {
+        console.log(error);
+        alert(error);
+      },
+    }
+  );
+
+  const handleShare = async (message: Message) => {
+    console.log(message);
+    try {
+      mutate({
+        name: chatUser,
+        prompt: message.text,
+        photo: `data:image/jpeg;base64,${message.image}`,
+        email: "",
+      });
+      setSharedImagesArr((prev) => [...prev, message.id]);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return (
+    <>
+      <div>
+        <form onSubmit={handleJoinChat}>
+          <input
+            type="text"
+            value={chatUser}
+            onChange={(e) => setChatUser(e.target.value)}
+            className="border-2 w-30 p-2"
+            placeholder="Type your message..."
+          />
+          <button className="bg-purple-500 hover:bg-purple-700 text-white font-bold py-2 px-4 rounded mt-2">
+            Join Chat
+          </button>
+        </form>
+      </div>
+      <div className="flex" style={{ height: "78vh" }}>
```

Comment:
Keep consistency using the tailwind classes.

```suggestion
      <div className="flex h-3/4">
```

Reply:
apparently, tailwind does not have a class for vh & vw, I had to add custom ones to Tailwind.config.

h-3/4 = height: 75%